examples/src/Syntax/ComposeFormula/Config.hs

```diff
@@ -21,7 +23,14 @@ small = ComposeFormulaConfig
     , maxDepth = 4
     , availableAtoms = "ABCD"
     , minAmountOfUniqueAtoms = 4
-    , allowArrowOperators = False
+    , operatorFrequencies = OperatorFrequencies
+      { and = 1
+      , or = 1
+      , impl = 0
+      , backImpl = 0
+      , equi = 0
+      , neg = 1
+      }
```

Comment:
The need to constantly hide the `and` and `or` functions seems a red flag to me.

Maybe it's anyway better to model this differently here in the datatype. How about something like a `Map BinOp Int` instead of the six different record fields?

If the config checker then also checks that `keys operatorFrequencies == [minBound .. maxBound :: BinOp]`, then lists like `[and, or, impl, backImpl, qui, neg]` occurring further below can also be written simpler as `elems operatorFrequencies`.

Reply:
Might also be worth considering to use something else than `Data.Map`, to actually guarantee that all `BinOp`s are contained. The package https://hackage.haskell.org/package/TotalMap seems dormant, but actually `Data.Array` should do well.

Reply:
What is the advantage of `Data.Array` over `Data.Map`? It does not seem that an array has to be total.

`TotalMap` only provides a few functions to access the data inside. `Data.Map` would be more versatile in that regard.

Reply:
> What is the advantage of `Data.Array` over `Data.Map`? It does not seem that an array has to be total.

I thought that `Array` would be "safer" since when initializing it with `(minBound, maxBound)` as the index range, one is assured that all entries are present. But admittedly, initializing a `Map` with `[minBound .. maxBound]` also achieves that. So there isn't really a more powerful static guarantee. One knows that there are no "gaps" in the keys, but an `Array` could still be initialized with `(succ minBound, maxBound)` and the type wouldn't tell that an entry is missing.

> `TotalMap` only provides a few functions to access the data inside.

Well, what API would be needed of the frequencies structure? Doesn't it boil down to just "accessing a certain element" and "returning all elements"? Then `getTotalMap` and `Foldable.toList` should suffice.

Reply:
I also used `Map.filter` and `Map.findWithDefault`. `Map.filter` could be replaced by the `Data.List.filter` implementation, `findWithDefault` would get obsolete. 

The final question would be if we really want to use `TotalMap` even though it might not be actively maintained anymore.

Reply:
Let's go with `Data.Map`.